app/controllers/api/v1/geocoding_controller.rb

```diff
@@ -36,7 +35,7 @@ def geocoding_data
   def address_within_neighborhood?(address)
```

Comment:
@chrisdolendo @creativecoder I feel like there should be a check before line 23 of this file (I can't leave a comment on line 23 because I didn't modify it in this commit) that checks whether the listing has ADHP or NRHP before calling `address_within_neighborhood`. My assumption here is that if the listing doesn't have either ADHP or NRHP, then the value for `boundary_match` is automatically nil because the concept of a match shouldn't apply. Do you know what the business rules are around this?

Reply:
@callaghanc I believe this was set up to always return `false` for boundary match because we use that somewhere in AngularJS to know if the geocoding request has been made for that address, or not, but I'd have to dig into the code to be sure.

If that is the case, I'd recommend adding a property to the response, like `"geocoded": true` and using that instead, because I think it would be more helpful if the Salesforce data we store didn't incorrectly store false matches that weren't relevant or were errors.

Reply:
@creativecoder This PR that is currently open and awaiting product acceptance: https://github.com/Exygy/sf-dahlia-web/pull/1030, affects the values returned for boundary match. It's a bit confusing because both this PR and that PR deal with the same bits of code, just doing slightly different things to them. Actually now I think of it this comment would have been better to leave on the other PR -__- but anyway, the changes in that other PR make it so the the boundary match now has three possible values it can return: true (to indicate a match), false (to indicate that the address was found not to be a match), and nil (to indicate that the address couldn't be put through the matching process due to error or blank address). So in the case of neither ADHP or NRHP being present, we could just return `boundary_match: nil` without even going into the `address_within_neighborhood?` function so that we can skip doing the whole matching process. But I am not sure how that meshes with the business rules.

Reply:
@callaghanc It's a good question to ask. The boundary matcher can still return results for the building's project ID, but I don't know what meaning this has if the listing doesn't have NRHP. You could ping on slack or bring up at stakeholder or design/product meeting.

Reply:
@creativecoder I'll do that, thanks for the suggestions of where to go with the q, that tends to be the fuzziest part for me.

Reply:
@creativecoder Yay Ashley answered my question and confirmed that the approach I mentioned above to boundary matching for listings that have neither NRHP nor ADHP is correct. I'm going to add the change with a comment to this PR. I also am going to add a test for the controller which holds all this logic. I don't think we have this knowledge documented anywhere so a test would be doubly useful for both validation and documentation.